### PR TITLE
[4.0] Change to use FontAwesome 5

### PR DESCRIPTION
**Pull request by brianteeman**
This is quite a big change and will need a composer and npm install to test

Many of the icons have changed name with FA5  see https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4#name-changes

The icomoon mapping has been updated with the name changes

Where possible I have updated any icons in core to the new names but I may have missed some.

### CODE REVIEW

**Comment by Hackwar** [comment]
Thank you for doing all this work. There is a CSS to keep on using FA4 names with FA5. Should we use this to keep the changes smaller? I also really, REALLY don't like that we can't use the Joomla icon in our own CMS from that icon font. Should we contact them and ask them to move it to the free group?

**Comment by mbabker** [comment]
> I also really, REALLY don't like that we can't use the Joomla icon in our own CMS from that icon font.

Umm, why?  I updated https://github.com/joomla/framework.joomla.org to use FA5 a while ago and use the Joomla icon there without issue, you just have to make sure you're loading the brand icons (none of which are in the pro group).

**Comment by brianteeman** [comment]
I must have misread the terms of the brand file.

As for the v4 to v5 shim @wilsonge correctly pointed out that joomla has never shipped a release with fa4

**Comment by brianteeman** [comment]
I will fix to include the brand if requested but for the sake of one icon I would prefer a different solution with less weight

**Comment by wilsonge** [comment]
For the Joomla brand icon let's create our own class that loads the `logo.svg` in the template https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/images/logo.svg

**Comment by brianteeman** [comment]
yes that's what i was thinking. that should be done in its own pr

**Comment by wilsonge** [comment]
The other option thinking about it is to load the brand icons in the backend template only. For things like the social media classes might be useful.

File: administrator/components/com_media/resources/scripts/components/tree/item.vue [hunk 1]
```diff
@@ -33,8 +33,8 @@
             iconClass() {
                 return {
                     fa: true,
-                    'fa-folder-o': !this.isOpen,
-                    'fa-folder-open-o': this.isOpen,
+                    'fa-folder': !this.isOpen,
+                    'fa-folderpen-o': this.isOpen,
```
Comment (Quy) [inline administrator/components/com_media/resources/scripts/components/tree/item.vue]:
```suggestion
                    'fa-folder-open': this.isOpen,
```

**Comment by brianteeman** [comment]
if the brand icons are included then it should be in both

**Comment by wilsonge** [comment]
Personally I don't think that's right. We can be more tight with the resources loaded in the frontend than the backend

**Comment by brianteeman** [comment]
The use case for social media icons its much greater in the front end

**Comment by mbabker** [comment]
If anything it's more common to load brand icons in the frontend than it is in the backend.

**Comment by dgrammatiko** [comment]
@wilsonge according to your decision ( https://github.com/joomla/40-backend-template/pull/441#issuecomment-400840436 ) the front end shouldn't have a dependency on FA. I hope you didn't change your mind on this.

PS. Also for this case to stand true all of the form fields **should** be font awesome free, as the fields are common to both sides frontend -backend...

**Comment by brianteeman** [comment]
I have added brands and updated the markup where necessary to use it (fa => fab)
It is a separate commit to can easily be removed

**Comment by richard67** [comment]
@brianteeman Shouldn't there be a change "fa" => "fas" for those icons in the standard package, e.g. change "fa fa-pen-square" to "fas fa-pen-square"? See [https://fontawesome.com/icons/pen-square?style=solid](https://fontawesome.com/icons/pen-square?style=solid)

That means for Fontawesome 5, the "fa" is in general changed to a three letter class where the last letter is the subpackage where it belongs to, i.e. "b" for brand, "s" for solid, "l" for light and "r" for regular.

**Comment by brianteeman** [comment]
no there is no need for that. FA5 supports both fa and fas

**Comment by richard67** [comment]
Hmm, their icon pages don't show that.

**Comment by brianteeman** [comment]
See https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4#changes

**Comment by richard67** [comment]
I see: "fas or fa = Font Awesome Solid". Thanks for the info. Seems I missed that in past.

**Comment by brianteeman** [comment]
@richard67 and of course I followed best practice and checked that it worked before I submitted the PR

**Comment by richard67** [comment]
@brianteeman I never assumed something else. But as we are all humans only, we all can make mistakes ;-)

**Comment by wilsonge** [comment]
That'll do for now :) We can definitely work on how we optimise things in frontend. But this is now just a straight version upgrade so we don't have to worry about that here

Thanks!

**Comment by brianteeman** [comment]
Thanks - learnt a lot writing this pr

**Comment by Scrabble96** [comment]
It's a shame, but fontawesome 4.7 fonts like file-o, which used to be white/transparent with a colour outline, have now been classed as 'light' in v 5.0 - and are now Pro, so no longer free. 

In Joomla 3x, in the template editor directory tree, for example, the file icon is light and easily distinguishable at a glance from from folders:

J3-white-icons

In Joomla 4.0 dev they are all solid:
j4-solid-icons

See pics of the different fontawesome options:

4.7 free:
fontawesome-4 7-file-o

5.0 free
fontawesome-5 0-free-file

5.0 Pro
fontawesome-5 0-pro-file

Is there a case for keeping both versions?

**Comment by ghost** [comment]
@Scrabble96 can you please open an new Issue (plus link to this) as Comments on closed one didn't get much notice.

**Comment by Scrabble96** [comment]
> 
> 
> @Scrabble96 can you please open an new Issue (plus link to this) as Comments on closed one didn't get much notice.

Hi, @franz-wohlkoenig. Thanks, I have just done so on issue #25915.

**Pull request merged**